src/model/entities/rundown.ts

```diff
@@ -635,4 +656,39 @@ export class Rundown extends BasicRundown {
   public getNextCursor(): RundownCursor | undefined {
     return this.nextCursor
   }
+
+  public replacePiece(pieceToBeReplaced: Piece, newPiece: Piece): void {
+    this.assertActive(this.replacePiece.name)
+    if (this.getActivePart().id === pieceToBeReplaced.getPartId()) {
+      this.getActivePart().replacePiece(pieceToBeReplaced, newPiece)
+      return
+    }
+
+    if (this.getNextPart().id === pieceToBeReplaced.getPartId()) {
+      this.getNextPart().replacePiece(pieceToBeReplaced, newPiece)
+      return
+    }
+
+    throw new UnsupportedOperation(`Can't replace Piece on Rundown ${this.id}. Piece ${pieceToBeReplaced.id} is neither on the active or next Part.`)
+  }
+
+  public getHistory(): Part[] {
+    return this.history
+  }
+
+  public findPartInHistory(predicate: (part: Part) => boolean): Part {
+    const historicPart: Part | undefined = [...this.history, this.getActivePart().clone()].reverse().find(predicate)
+    if (!historicPart) {
+      throw new NoPartInHistoryException(`Rundown ${this.id} does not contain a Part with the specified conditions in its history`)
+    }
+    return historicPart
+  }
```

Comment:
How does it make it inflexible?
But just because a thing is inflexible does not necessarily mean it is a bad thing (it often does, but not always). Immuteability is inflexible, but we still want it ;) 

It could be argued it's a good thing that it is the Rundown who is responsible for searching the history. Afterall, it is the history of the Rundown.

I think this should be something decided on in a different task :)

Reply:
If I have a scenario, where I want to find the first occurrence of a given part in the history, then I can't use this function. Or if I want to find a part, in relation to other parts in the history, then that is neither possible.

And yes, it is the history of the rundown, but the responsibility that the rundown has for the history is maintaining it and ensuring that it can only be accessed in the right way (e.g. compositing the structure and/or making it readonly). One way to access it is to do a search like here, but it is tailored for the specific situation that we need it for.

Another thing is that `.getHistory()` returns a mutable reference, such that the caller can alter the internal history of the rundown #timeTravel.

How is immutability inflexible? :D

I am fine with it being changed in another task, but which task?

Reply:
The task we create for it? ;)